forms/HeaderField.php

```diff
@@ -17,31 +17,17 @@ class HeaderField extends DatalessField {
 	 */
 	protected $headingLevel = 2;
 
+	protected $schemaDataType = self::SCHEMA_DATA_TYPE_STRUCTURAL;
+
+	protected $schemaComponent = 'HeaderField';
+
 	/**
 	 * @param string $name
-	 * @param null|string $title
+	 * @param string $title
 	 * @param int $headingLevel
 	 */
-	public function __construct($name, $title = null, $headingLevel = 2) {
```

Comment:
probably @tractorcow would be better to answer this, he mentioned that keeping legacy functionality like this isn't ideal?

I had found it using $name as the title if the original title passed in ended up being a numeric value (which can happen, e.g. the case of file names)
originally, I had a `count($args) < 3` check, and maybe even a `is_numeric($args[1]) && $args[1] < 6` would probably suffice